### DISCUSSION TITLE
Sourcehut integration

### DISCUSSION THREAD

**Discussion by TechnologyClassroom** [woodpecker-ci/woodpecker#2288] [Request a Feature]
### Clear and concise description of the problem

It would be cool to have the option to use Woodpecker CI for the [sourcehut](https://sourcehut.org/) forge.

### Suggested solution

Not familiar enough with Woodpecker to make a suggested solution.

### Alternative

Sourcehut has its own CI service, but Woodpecker seems to have more features.  https://man.sr.ht/builds.sr.ht/

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/woodpecker-ci/woodpecker/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://woodpecker-ci.org/docs/intro).
- [X] Check that there isn't already an [issue](https://github.com/woodpecker-ci/woodpecker/issues) that request the same feature to avoid creating a duplicate.

**Comment by 6543** [comment]
the core team has other prios ... but if somebody like to pick this issue up ...
he just have to look into https://github.com/woodpecker-ci/woodpecker/tree/master/server/remote

if he can can cover the remote interface with a SH specific impl it's done ...

**Comment by mwbrown** [comment]
(sorry to necro this, I know it's been over three years :smile:)

Hey! I was actually just looking into this today because I'm also interested in using Woodpecker with Sourcehut, however, based on my reading of the Forge interface requirements and what Sourcehut supports, the job looks like it might be a bit more involved. Sourcehut has both an OAuth 2.0 provider as well as a GraphQL API, but it looks like they might only support checking out private repos using SSH.

If we wanted to add this to Woodpecker, it might involve either generating SSH keys to attach to the `User` and uploading them via the API, or we have some other way to specify creds to the CI server so it can clone the repository.

I'm happy to keep looking into this if we think it's feasible, I know @6543 pointed to the remote API which right now looks like it's called Custom Forge on the current docs. The tricky bit would just be cloning the repo during a build.
